deckhouse-controller/pkg/controller/module-controllers/release/controller.go

```diff
@@ -701,6 +710,15 @@ func isModuleExistsOnFS(symlinksDir, symlinkPath, modulePath string) bool {
 	return targetPath == modulePath
 }
 
+func moduleRegistryConfigChanged(mr *v1alpha1.ModuleRelease) bool {
```

Comment:
The method name is `module registry config changed`, but it checks whether the restart required annotation is present inside. And there is also a side effect: it returns the boolean value and deletes the annotation.